File: 0228-summary-ranges/0228-summary-ranges.py
class Solution:
    def summaryRanges(self, nums: List[int]) -> List[str]:
        result = []
        start = 0
        for i in range(0, len(nums)):
            end = i
            if i == len(nums)-1 or nums[i] != nums[i+1]-1:
                if start == end:
                    result.append(str(nums[start]))
                else:
                    result.append(str(nums[start]) + "->" + str(nums[end]))
                start = i+1
        return result
                    

File: 0228-summary-ranges/0228-summary-ranges.java
class Solution {
    public List<String> summaryRanges(int[] nums) {
        List<String> result = new ArrayList<>();
        int start = 0;
        for(int i=0;i<nums.length;i++){
             int end = i;
            if(i == nums.length-1 || nums[i] != nums[i+1]-1){
                if(start == end){
                    result.add(String.valueOf(nums[start]));
                } else {
                    result.add(nums[start] + "->" + nums[end]);
                }
                start = i+1;
            } 
        }
        return result;
    }
}